### PR TITLE
Add pycodestyle to CI and fix all code formatting errors

### LINKED ISSUE #559: Add pycodestyle to CI and fix all code formatting errors

### PR DESCRIPTION
**Pull request by clinssen**
Fixes #559.

Enhance Travis CI suite with:
- copyright header check for all Python source files: check consistent use of copyright header template
- pycodestyle check for all Python source files: check that all conform to PEP8 guidelines on whitespace, no bare excepts, etc.

### CODE REVIEW

File: extras/codeanalysis/check_copyright_headers.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""
+Script to check if all files have a proper copyright header.
+
+This script checks the copyright headers of all C/C++/SLI/Python files
+in the source code against the corresponding templates defined in
+"doc/copyright_header_template.*".
```
Comment (clinssen) [inline extras/codeanalysis/check_copyright_headers.py:30]:
```suggestion
"extras/codeanalysis/copyright_header_template.*".
```

File: extras/codeanalysis/check_copyright_headers.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""
+Script to check if all files have a proper copyright header.
+
+This script checks the copyright headers of all C/C++/SLI/Python files
```
Comment (clinssen) [inline extras/codeanalysis/check_copyright_headers.py:28]:
```suggestion
This script checks the copyright headers of all Python files
```

**Comment by clinssen** [comment]
@jougs: I went through the complete diff on GitHub's web interface, and it all passes the sanity check. If you would base your review mostly on the changes in ``.travis.yml`` and ``extras/codeanalysis/check_copyright_headers.py``, and perhaps skim the diff of the other files, I'm sure that would be adequate. Thanks in advance!

File: extras/codeanalysis/check_copyright_headers.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+#
+# -*- coding: utf-8 -*-
+#
+# check_copyright_headers.py
+#
+# This file is part of NEST.
```
Comment (jougs) [inline extras/codeanalysis/check_copyright_headers.py:7]:
```suggestion
# This file is part of NESTML.
```
Or will this change be part of another PR?

**Review by jougs — COMMENTED** [review]
Didn't we agree to use NESTML instead of NEST in the copyright header? Or am I confused? Thanks!

File: extras/codeanalysis/check_copyright_headers.py (hunk 3 above)
Comment (clinssen) [reply extras/codeanalysis/check_copyright_headers.py:7]: Yes, we're still waiting for HEP's blessing, perhaps we can discuss it in the NEST meeting today. This PR is only making things consistent for now without introducing that change yet.

**Review by jougs — APPROVED** [review]
